services/src/main/java/org/keycloak/forms/login/freemarker/model/AbstractUserProfileBean.java

```diff
@@ -52,12 +53,34 @@ protected void init(KeycloakSession session, boolean writeableOnly) {
     protected abstract UserProfile createUserProfile(UserProfileProvider provider);
 
     /**
-     * Get attribute default value to be pre-filled into the form on first show.
+     * Get attribute default value to be pre-filled into the form on first show - for fields which expect one value only.
      * 
      * @param name of the attribute
      * @return attribute default value (can be null)
+     * @see #getAttributeDefaultValues(String)
      */
-    protected abstract String getAttributeDefaultValue(String name);
+    protected String getAttributeDefaultValue(String name) {
+        Stream<String> vs = getAttributeDefaultValues(name);
+        if(vs == null) {
+            return null;
+        } else {
+            List<String> al = vs.collect(Collectors.toList());
+            if(al == null || al.isEmpty())
+                return null;
+            else
+                return al.get(0);
+            // next stream based code should be better, but it produces NPE in some cases :-(
```

Comment:
You should probably filter null values as `findFirst` should throw NPE if the selected element is null. 

Also, this method seems unnecessary so we can just change `getValue` to use the new method that returns the stream (or perhaps just rely on the new `getValues` method)?

Reply:
> You should probably filter null values as `findFirst` should throw NPE if the selected element is null.

You are right, it is behavior of `findFirst` documented in javadoc to throw NPE if the first element is null value. But in this case we can't use it as we should work with values from the stream here as is. 
 
> Also, this method seems unnecessary so we can just change `getValue` to use the new method that returns the stream (or perhaps just rely on the new `getValues` method)?

Right, it will be much simpler code to base `getValue` on `getValues`.